src/relay/backend/build_module.cc

```diff
@@ -300,6 +300,7 @@ class RelayBuildModule : public runtime::ModuleNode {
       }
     });
     pass_seqs.push_back(transform::EliminateCommonSubexpr(fskip));
+    pass_seqs.push_back(transform::InferType());
```

Comment:
Do we really need this? InferType is already in the requirement of SimplifyExpr. Other existing patterns also refer to checked_type_ so it should be fine.

Reply:
Emmm... I dont know why Ci fails, just try it

Reply:
From the CI it does look like the type is missing. Since InferType is executed before and after SimplifyExpr, one possibility is your `data` is already mutated by previous patterns in this pass and that callback didn't manually specify the type.